internal/resources/metal/virtual_circuit/resource.go

```diff
@@ -0,0 +1,459 @@
+			vc, resp, err := client.InterconnectionsApi.GetVirtualCircuit(ctx, id).Execute()
+			if err != nil {
+				if resp != nil {
+					err = equinix_errors.FriendlyErrorForMetalGo(err, resp)
+				}
+				return 0, "", err
+			}
+			vcStatus := ""
+			if vc.VlanVirtualCircuit != nil {
+				vcStatus = string(vc.VlanVirtualCircuit.GetStatus())
+			} else {
+				vcStatus = string(vc.VrfVirtualCircuit.GetStatus())
+			}
+			return vc, vcStatus, nil
+		},
+		Timeout:    timeout,
+		Delay:      10 * time.Second,
+		MinTimeout: 5 * time.Second,
+	}
+}
+
+func resourceMetalVirtualCircuitUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).NewMetalClientForSDK(d)
+	needsUpdate := false
+
+	ur := metalv1.VirtualCircuitUpdateInput{}
+
+	if _, ok := d.GetOk("vlan_id"); ok {
+		ur.VlanVirtualCircuitUpdateInput = &metalv1.VlanVirtualCircuitUpdateInput{}
+		if d.HasChange("vlan_id") {
+			needsUpdate = true
+			vnid := d.Get("vlan_id").(string)
+			ur.VlanVirtualCircuitUpdateInput.Vnid = &vnid
+		}
+
+		if d.HasChange("name") {
+			needsUpdate = true
+			name := d.Get("name").(string)
+			ur.VlanVirtualCircuitUpdateInput.Name = &name
+		}
+
+		if d.HasChange("description") {
+			needsUpdate = true
+			desc := d.Get("description").(string)
+			ur.VlanVirtualCircuitUpdateInput.Description = &desc
+		}
+
+		if d.HasChange("speed") {
+			needsUpdate = true
+			speed := d.Get("speed").(string)
+			ur.VlanVirtualCircuitUpdateInput.Speed = &speed
+		}
+
+		if d.HasChange("tags") {
+			needsUpdate = true
+			ts := d.Get("tags")
+			sts := []string{}
+
+			switch ts.(type) {
+			case []interface{}:
+				for _, v := range ts.([]interface{}) {
+					sts = append(sts, v.(string))
+				}
+				ur.VlanVirtualCircuitUpdateInput.Tags = sts
+			default:
+				return diag.Errorf("garbage in tags: %s", ts)
+			}
+		}
+	} else {
+		ur.VrfVirtualCircuitUpdateInput = &metalv1.VrfVirtualCircuitUpdateInput{}
+
+		if d.HasChange("name") {
+			needsUpdate = true
+			name := d.Get("name").(string)
+			ur.VrfVirtualCircuitUpdateInput.Name = &name
+		}
+
+		if d.HasChange("description") {
+			needsUpdate = true
+			desc := d.Get("description").(string)
+			ur.VrfVirtualCircuitUpdateInput.Description = &desc
+		}
+
+		if d.HasChange("speed") {
+			needsUpdate = true
+			speed := d.Get("speed").(string)
+			ur.VrfVirtualCircuitUpdateInput.Speed = &speed
+		}
+
+		if d.HasChange("tags") {
+			needsUpdate = true
+			ts := d.Get("tags")
+			sts := []string{}
+
+			switch ts.(type) {
+			case []interface{}:
+				for _, v := range ts.([]interface{}) {
+					sts = append(sts, v.(string))
+				}
+				ur.VrfVirtualCircuitUpdateInput.Tags = sts
+			default:
+				return diag.Errorf("garbage in tags: %s", ts)
+			}
+		}
+	}
+
+	if needsUpdate {
+		if _, _, err := client.InterconnectionsApi.UpdateVirtualCircuit(ctx, d.Id()).VirtualCircuitUpdateInput(ur).Execute(); err != nil {
+			return diag.FromErr(equinix_errors.FriendlyError(err))
```

Comment:
Should this be: `equinix_errors.FriendlyErrorForMetalGo`? There are a few other instances in this PR.

Reply:
IMO it should just be `diag.FromErr(err)`

Reply:
I didn't check all of the uses. I was considering how we noticed FriendlyError vs. FriendlyErrorForMetalGo usage in other locations, which could potentially lead to improper 404 detection.

Reply:
I changed the `FriendlyError` calls to remove the conversion entirely, since in one case there's no status check happening and in the other cases the error has already been (begrudgingly) converted in order to fit into the existing filtering and avoid a larger refactor.